crates/dprint/src/cli/run_cli.rs

```diff
@@ -266,29 +266,69 @@ fn run_editor_service<TEnvironment: Environment>(
     let stdout = environment.stdout();
     let reader_writer = StdIoReaderWriter::new(stdin, stdout);
     let mut messenger = StdIoMessenger::new(reader_writer);
+    let config = resolve_config_from_args(&args, cache, environment)?;
+    let (mut file_patterns, mut absolute_paths) = get_config_file_paths(&config, args, environment)?;
+    let mut file_paths = resolve_file_paths(&file_patterns, &absolute_paths, &config, environment)?;
     let mut past_config: Option<ResolvedConfig> = None;
 
+    fn handle_check_path<TEnvironment: Environment>(
+        messenger: &mut StdIoMessenger<Box<dyn std::io::Read + Send>, Box<dyn std::io::Write + Send>>,
```

Comment:
Wasn't sure if there was a better way to pass these vars in -- I did try with a closure but handling the lifetimes because a little annoying.

Reply:
This is ok! Yeah, sometimes when this happens it can help to either store all the state in a struct and then the functions can act on that struct.

For example, this could maybe all be extracted out to an `EditorService` struct that has a `start()` function on it or something. That would store all this state on the `EditorService` struct then `handle_check_path` could be a private implementation function on it. Anyway, don't worry about changing that in this PR as it's a bigger refactor that could maybe come down the line.